src/java/org/apache/cassandra/auth/CassandraRoleManager.java

```diff
@@ -172,14 +172,14 @@ protected final void loadRoleStatement()
     public void createRole(AuthenticatedUser performer, RoleResource role, RoleOptions options)
     throws RequestValidationException, RequestExecutionException
     {
-        String insertCql = options.getPassword().isPresent()
+        String insertCql = options.getPassword().isPresent() || options.getHashedPassword().isPresent()
                          ? String.format("INSERT INTO %s.%s (role, is_superuser, can_login, salted_hash) VALUES ('%s', %s, %s, '%s')",
                                          SchemaConstants.AUTH_KEYSPACE_NAME,
                                          AuthKeyspace.ROLES,
                                          escape(role.getRoleName()),
                                          options.getSuperuser().or(false),
                                          options.getLogin().or(false),
-                                         escape(hashpw(options.getPassword().get())))
+                                         options.getHashedPassword().or(() -> escape(hashpw(options.getPassword().get()))))
```

Comment:
Guava's `Optional#or` has been marked as beta for a while. I think that `RoleOptions` uses Guava's `Optional` because it was created before Java 8, now we can easily replace it with `java.util.Optional`. I gave it a try [here](https://github.com/adelapena/cassandra/commit/f6bf02fd02aa58ae99f15b4865a212204d2ee907) and the change seems pretty straightforward. I know this is not strictly related to the patch but since we are here...

Reply:
sgtm